design/20241124-rename-bunde-to-clusterbundle.md

```diff
@@ -0,0 +1,158 @@
+for trust bundle management inside namespaces in multi-tenant clusters.
+
+In addition, `ClusterBundle` would be a better name looking at cert-manager having `Issuer` and `ClusterIssuer`.
+The name would also be better aligned with upstream [ClusterTrustBundle](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#cluster-trust-bundles).
+
+### Goals
+
+- `Bundle` resource is renamed to `ClusterBundle`
+- Provide a smooth migration for users of trust-manager
+
+
+### Non-Goals
+
+- Change/fix/improve `Bundle`/`ClusterBundle` API while renaming
+- Introduce a new API version
+- Introduce a namespace-scoped bundle API (at least not now)
+
+## Proposal
+
+We propose to do this change over multiple releases:
+
+1. Version X: Only `Bundle` API is served and has the functional controller. (present state)
+2. Version X+1: We introduce `ClusterBundle` with API otherwise identical to `Bundle`:
+   - `Bundle` is marked as deprecated.
+   - Functional controller is migrated to `ClusterBundle`
+   - Validating webhook for both `Bundle` and `ClusterBundle` with same validations
+   - New technical controller added for `Bundle`, converting bundles to cluster bundles
+3. Version Y: Remove `Bundle` and only serve `ClusterBundle` (desired state)
+
+Some of these actions/states require extra attention, which are further described below.
+
+### Functional Controller Migration
+
+This should be rather simple. Since the APIs are identical, we just have to perform a "smart"
+search and replace in the controller code. When the controller initially boots up on the X+1 version,
+there will be no `ClusterBundle` resources in the cluster, as these will be created/updated by the technical
+controller.
+
+The `ClusterBundle` reconciler should do the same as for `Bundle` in version X, but we should test/verify that
+the controller is able to change owner references on target resources from `Bundle` to `ClusterBundle` without
+facing issues.
+
+We must also ensure that the `ClusterBundle` reconciler isn't deleting target resources it thinks are "orphaned".
+We believe the controller will only delete target resources if/when the controller resource exists, and
+otherwise rely on the Kubernetes garbage collector to delete orphans. But this should be tested/verified.
+
+### Temporary Technical Controller
+
+When the functional controller is migrated to target `ClusterBundle`, we want to introduce a temporary technical
+controller targeting the now deprecated `Bundle` resource.
+
+This controller should just create or update the `ClusterBundle` with the same name as the reconciled `Bundle`.
+Since the APIs should have identical `spec`, this should be a simple controller.
+
+To allow a user to migrate their resources from `Bundle` to `ClusterBundle`, the controller should not act
+on `Bundle` delete events. Nor add an owner reference to `ClusterBundle` to avoid the Kubernetes garbage collector
+to act on deletes.
```

Comment:
nitpick (wording): (it really doesn't matter though, it's perfectly understandable as is!)

```suggestion
on `Bundle` delete events - nor should it add an owner reference to `ClusterBundle` to avoid the Kubernetes garbage collector
acting on deletes.
```